### PR TITLE
Render helpstring for incomplete subcommands

### LINKED ISSUE #487: 🐛 BUG: render helpstring for incomplete subcommands
### What version of `Wrangler` are you using?

0.0.17

### What operating system are you using?

Linux

### Describe the Bug

this isn't a bug, but a UX improvement from wrangler1 that we should bring over to wrangler2.
the happy-path is the same, but wrangler1 prints helptext when a subcommand is incomplete, whereas wrangler2 does nothing.
eg:
```
$ wrangler1 secret list
[]
$ wrangler2 secret list
[]
$ wrangler1 secret 
wrangler-secret 1.19.8
Generate a secret that can be referenced in the worker script

USAGE:
    wrangler secret [FLAGS] [OPTIONS] <SUBCOMMAND>

FLAGS:
    -h, --help       Prints help information
        --verbose    Toggle verbose output (when applicable)

OPTIONS:
    -c, --config <config>    Path to configuration file [default: wrangler.toml]
    -e, --env <env>          Environment to perform a command on

SUBCOMMANDS:
    delete    Delete a secret variable from a script
    help      Prints this message or the help of the given subcommand(s)
    list      List all secrets for a script
    put       Create or update a secret variable for a script
$ wrangler2 secret 
$
```

wrangler2 has the helptext though, so we should render it
```
$ wrangler2 secret -h
wrangler secret

🤫 Generate a secret that can be referenced in the worker script

Commands:
  wrangler secret put <key>     Create or update a secret variable for a script
  wrangler secret delete <key>  Delete a secret variable from a script
  wrangler secret list          List all secrets for a script

Flags:
  -c, --config   Path to .toml configuration file  [string]
  -h, --help     Show help  [boolean]
  -v, --version  Show version number  [boolean]

Options:
  -l, --local  Run on my machine  [boolean] [default: false]
```

I can try to fix this when I have time, but feel free to patch it before then if you know how :)

### PR DESCRIPTION
**Pull request by JacobMGEvans**
When a command with subcommands is ran without subcommands the help message should display.

resolves #487

### CODE REVIEW

File: packages/wrangler/src/index.tsx [hunk 1]
```diff
@@ -2186,77 +2199,81 @@ export async function main(argv: string[]): Promise<void> {
     }
   );
 
-  wrangler.command("pages", "⚡️ Configure Cloudflare Pages", pages);
+  wrangler.command("pages", "⚡️ Configure Cloudflare Pages", (pagesYargs) =>
+    pages({ pagesYargs, subHelp } as unknown as Argv<unknown>)
```
Comment (petebacondarwin) [inline packages/wrangler/src/index.tsx:2203]: This is a bit magical. Can we just do something like:

```ts
wrangler.command("pages", "...", (pagesYargs) => {
  return pages(pargesYargs).command(subHelp);
}
```

File: packages/wrangler/src/__tests__/index.test.ts [hunk 2]
```diff
@@ -863,4 +863,111 @@ describe("wrangler", () => {
             `);
     });
   });
+
+  describe("subcommand implicit help ran on imcomplete command execution", () => {
+    function endEventLoop() {
+      return new Promise((resolve) => setImmediate(resolve));
+    }
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/index.test.ts:870]: Clever!

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Nice work @JacobMGEvans. I left a suggestion, which I have not tested, on how we can avoid messing with the pages stuff.

File: packages/wrangler/src/index.tsx (hunk 1 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/index.tsx:2203]: I'll give that a try 👍

File: packages/wrangler/src/__tests__/index.test.ts (hunk 2 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/__tests__/index.test.ts:870]: Thanks! 😆

File: packages/wrangler/src/index.tsx (hunk 1 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/index.tsx]: Was able to get it working with this`pages(pagesYargs.command(subHelp))` 😄

**Review by petebacondarwin — APPROVED** [review]
Nice work @JacobMGEvans

**Pull request merged**